src/controller.ts

```diff
@@ -1,2 +1,24 @@
-console.error('Not implemented')
-process.exit(1)
+#!/usr/bin/env node
+
+import { supplyClient } from './ZeroconfClient'
+import Client from './Client'
+
+import * as bonjour from 'bonjour'
+import { once } from 'events'
+
+const zeroconf = bonjour()
+
+function done() {
+	console.log('Finished')
+
+	// The mDNS socket apparently has no way to tell that it's not needed.
+	zeroconf.destroy()
+}
+
+const client = new Client(['fifth', 'fourth', 'third', 'second', 'first'])
```

Comment:
How would this work if someone wants multiple clients? Would there be multiple instances of this controller running?

Reply:
Sure, how you use the `Client` is up to you ("you" being the user of the client library). I think the controller would probably just use one client and exit when it completes, but it's flexible. (instances of the client class are independent from each other).